daemon/cluster/cluster.go

```diff
@@ -1108,7 +1108,7 @@ func (c *Cluster) CreateService(s types.ServiceSpec, encodedAuth string) (*apity
 		if err != nil {
 			logrus.Warnf("unable to pin image %s to digest: %s", ctnr.Image, err.Error())
 			resp.Warnings = append(resp.Warnings, fmt.Sprintf("unable to pin image %s to digest: %s", ctnr.Image, err.Error()))
-		} else {
+		} else if ctnr.Image != digestImage {
```

Comment:
Should we leave this as it is? Maybe it is worthwhile to have a log message specifying which digest the service was pinned to upon creation, even if the client provided a digest to begin with.